pkg/build/paths.go

```diff
@@ -33,28 +30,68 @@ type Paths struct {
 	MetaDir string
 }
 
-func NewBuildPaths(root, executableName, zipName string) (Paths, error) {
+type pathsSettings struct {
+	// targetDir is used to calculate the default bin path.
+	targetDir string
+}
+
+// buildPathsSettings contains optional settings for build paths.
+type buildPathsSettings struct {
+	targetDir string
+}
+
+type BuildPathsOpt func(s *pathsSettings)
+
+func WithTargetDir(path string) BuildPathsOpt {
+	return func(s *pathsSettings) { s.targetDir = path }
+}
+
+func NewBuildPaths(root, executableName, zipName string, opts ...BuildPathsOpt) (Paths, error) {
 	var bp Paths
 	if !filepath.IsAbs(root) {
 		return bp, fmt.Errorf("root path %q is not absolute", root)
 	}
-	return bp.setDefaults(root, executableName, zipName), nil
+
+	settings := pathsSettings{}
+	// Apply options.
+	for _, o := range opts {
+		o(&settings)
+	}
+
+	return bp.setDefaults(root, executableName, zipName, settings), nil
 }
 
 func (bp Paths) ZipDir() string {
 	return filepath.Dir(bp.ZipPath)
 }
 
+// TargetDir is the absolute path to the dir where any other files
+// needed to be included in the zip file should be placed.
+func (bp Paths) TargetDir() string {
+	return filepath.Dir(bp.BinPath)
+}
```

Comment:
This function replaces the original struct field, thus enforcing the invariant and avoiding the need to validate.